lib/msf/core/exploit/remote/http/nagios_xi/login.rb

```diff
@@ -0,0 +1,89 @@
+# -*- coding: binary -*-
+
+module Msf::Exploit::Remote::HTTP::NagiosXi::Login
+  # performs a Naxios XI login
+  #
+  # @param user [String] Username
+  # @param pass [String] Password
+  # @return [String, Msf::Exploit::CheckCode] String containing the session cookies on successful login, Msf::Exploit::CheckCode otherwise
+  def nagios_xi_login(user, pass)
+    # empty the cookie jar, otherwise the module will start failing if it is run multiple times in a short timespan
+    self.cookie_jar = [].to_set
+
+    # visit the login page in order to obtain the cookies and the `nsp_str` token required for authentication
+    res_pre_login = send_request_cgi({
+      'method' => 'GET',
+      'uri' => nagios_xi_login_url,
+      'keep_cookies' => true,
+    })
+
+    unless res_pre_login
+      return Msf::Exploit::CheckCode::Unknown('Connection failed')
+    end
+
+    unless res_pre_login.code == 200 && res_pre_login.body.include?('<b>Nagios XI</b>')
+      return Msf::Exploit::CheckCode::Safe('Target is not a Nagios XI application')
+    end
+
+    nsp = get_nsp(res_pre_login)
+
+    if nsp.blank?
+      return Msf::Exploit::CheckCode::Unknown('Unable to obtain the value of the `nsp_str` token required for authentication')
+    end
```

Comment:
This check failed on a (very) old version of Nagios XI (version 2012R2.9). I haven't bothered to look into why.

Here's the HTML.

<details><p>

```
msf6 auxiliary(scanner/http/nagios_xi_scanner) > run

####################
# Request:
####################
GET /nagiosxi/login.php HTTP/1.1
Host: 10.1.1.111
User-Agent: Mozilla/4.0 (compatible; MSIE 6.0; Windows NT 5.1)


####################
# Response:
####################
HTTP/1.1 200 OK
Date: Sat, 06 Feb 2021 03:29:29 GMT
Server: Apache/2.2.15 (CentOS)
X-Powered-By: PHP/5.3.3
Set-Cookie: nagiosxi=uk2g58fkahi5q65h0ki2n7b497; expires=Sat, 06-Feb-2021 03:59:29 GMT; path=/
Expires: Thu, 19 Nov 1981 08:52:00 GMT
Cache-Control: no-store, no-cache, must-revalidate, post-check=0, pre-check=0
Pragma: no-cache
Connection: close
Transfer-Encoding: chunked
Content-Type: text/html; charset=UTF-8

<!DOCTYPE html>
<!-- <!DOCTYPE html PUBLIC "-//W3C//DTD XHTML 1.0 Frameset//EN" "http://www.w3.org/TR/xhtml1/DTD/xhtml1-frameset.dtd"> -->
<html>
	
	<head>	
    <meta http-equiv="X-UA-Compatible" content="IE=Edge" />
	<!-- Produced by Nagios XI.  Copyyright (c) 2008-2011 Nagios Enterprises, LLC (www.nagios.com). All Rights Reserved. -->
	<!-- Powered by the Nagios Synthesis Framework -->
	<title>Nagios XI - Login</title>
	<meta name="ROBOTS" content="NOINDEX, NOFOLLOW">
	<meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />

	
		<link rel="shortcut icon" href="http://10.1.1.111/nagiosxi/images/favicon.ico" type="image/ico" />
	<link rel='stylesheet' type='text/css' href='http://10.1.1.111/nagiosxi/includes/css/jquery.autocomplete.css' />
	
	<script type='text/javascript'>
	//javascript:alert(document.documentMode);
	
		var base_url="http://10.1.1.111/nagiosxi/";
		var backend_url="http%3A%2F%2F10.1.1.111%2Fnagiosxi%2Flogin.php";
		var ajax_helper_url="http://10.1.1.111/nagiosxi/ajaxhelper.php";
		var ajax_proxy_url="http://10.1.1.111/nagiosxi/ajaxproxy.php";
		var suggest_url="http://10.1.1.111/nagiosxi/suggest.php";
		var request_uri="%2Fnagiosxi%2Flogin.php";
		var permalink_base="http://10.1.1.111/nagiosxi/login.php?";
		var demo_mode=0;
		var nsp_str="fb3798156288cc818e648fbf2725464a";
	</script>

<!-- FIREBUG LITE! -->	
<!--
<script type='text/javascript' 
        src='http://getfirebug.com/releases/lite/1.2/firebug-lite-compressed.js'></script>
//-->	


	
	<!-- main jquery libraries --> 
	<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/jquery/jquery-1.8.2.min.js?2012R2.9'></script>
	<link type="text/css" href="http://10.1.1.111/nagiosxi/includes/js/jquery/css/smoothness/jquery-ui-1.9.0.custom.min.css?2012R2.9" rel="stylesheet" />	
	<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/jquery/jquery.colorBlend.js'></script>	
	<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/jquery/jquery.timers-1.1.3.js'></script>	
	<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/jquery/jquery-ui-1.9.0.custom.min.js?2012R2.9'></script>

	<!-- DEPRECATED JQUERY PLUGINS
	<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/jquery/jquery.autocomplete.js'></script>
	<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/jquery/jquery.checkboxes.js'></script> 
	<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/jquery/jquery.bgiframe.pack.js'></script>
	<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/jquery/jquery.tooltip.pack.js'></script>
	<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/jquery/jquery.sparkline.js'></script>	
	<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/jquery/jquery.inview.min.js'></script>	
	-->
		
	<!-- colorpicker -->
	<link rel="stylesheet" href="http://10.1.1.111/nagiosxi/includes/js/jquery/colorpicker/css/colorpicker.css" type="text/css" />
	<script type="text/javascript" src="http://10.1.1.111/nagiosxi/includes/js/jquery/colorpicker/js/colorpicker.js"></script>
	
	<!-- clipboard plugin -->
	<script type="text/javascript" src="http://10.1.1.111/nagiosxi/includes/js/jquery/jquery.zclip.min.js"></script>

	<!-- XI JS Scripts -->
		<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/core.js?2012R2.9'></script>
		<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/commands.js?2012R2.9'></script>
	<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/views.js?2012R2.9'></script>
	<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/dashboards.js?2012R2.9'></script>
	<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/dashlets.js?2012R2.9'></script>
	<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/tables.js?2012R2.9'></script>
	<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/users.js?2012R2.9'></script>
	<script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/perfdata.js?2012R2.9'></script>
    <script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/wizards.js?2012R2.9'></script>
	
	<!-- XI CSS --> 
		<link rel='stylesheet' type='text/css' href='http://10.1.1.111/nagiosxi/includes/css/nagiosxi.css?2012R2.9' />
    
    <!-- Highcharts Graphing Library -->
    <script type="text/javascript" src="http://10.1.1.111/nagiosxi//includes/js/highcharts/highcharts.js?320"></script>
          <script type="text/javascript" src="http://10.1.1.111/nagiosxi//includes/js/highcharts/modules/exporting.js?320"></script><script type="text/javascript" src="http://10.1.1.111/nagiosxi//includes/js/highcharts/themes/gray.js?320"></script>
    <!-- D3 Graphing Library -->
    <script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/js/d3/d3.v3.min.js?2012R2.9'></script>
	
<!-- styles needed by jScrollPane -->
<link type="text/css" href="http://10.1.1.111/nagiosxi/includes/js/jquery/css/jquery.jscrollpane.css" rel="stylesheet" media="all" />

<!-- the jScrollPane script -->
<script type="text/javascript" src="http://10.1.1.111/nagiosxi/includes/js/jquery/jquery.jscrollpane.min.js"></script>

<!-- the mousewheel plugin - optional to provide mousewheel support -->
<script type="text/javascript" src="http://10.1.1.111/nagiosxi/includes/js/jquery/jquery.mousewheel.js"></script>

	
<link rel='stylesheet' type='text/css' href='http://10.1.1.111/nagiosxi/includes/dashlets/internethealthreport/internethealthreport.css' />
<link rel='stylesheet' type='text/css' href='http://10.1.1.111/nagiosxi/includes/dashlets/internettrafficreport/internettrafficreport.css' />
<link rel='stylesheet' type='text/css' href='http://10.1.1.111/nagiosxi/includes/dashlets/rss_dashlet/rss_dashlet.css' />
<link rel='stylesheet' type='text/css' href='http://10.1.1.111/nagiosxi/includes/dashlets/sansrisingports/sansrisingports.css' />
<link rel='stylesheet' type='text/css' href='http://10.1.1.111/nagiosxi/includes/dashlets/worldtimeserver/worldtimeserver.css' />
		<link rel='stylesheet' type='text/css' href='http://10.1.1.111/nagiosxi/includes/components/ccm/css/style.css?2.1' /><script type="text/javascript" src="http://10.1.1.111/nagiosxi/includes/components/ccm/javascript/main_js.js?2.1"></script><style type="text/css">
#contentWrapper { margin: 0px auto; width: 95%; } 
</style>
<script type="text/javascript">
var NAGIOSXI=true
</script><script type="text/javascript" src="http://10.1.1.111/nagiosxi/includes/components/graphexplorer/includes/graphexplorerinclude.js"></script><script type='text/javascript' src='http://10.1.1.111/nagiosxi/includes/components/helpsystem/helpsysteminclude.js?0.3'></script><link rel='stylesheet' type='text/css' href='http://10.1.1.111/nagiosxi/includes/components/helpsystem/helpsystem.css?0.3' /></head>


<body  class=' parent' >

		
	<div  class="parentpage"><!-- page-->

		<div id="header" class="parenthead" >

<!--- HEADER START -->


<div id="toplogo">
   <a href="http://www.nagios.com/products/nagiosxi/" target="_blank"><img src="http://10.1.1.111/nagiosxi/images/nagiosxi-logo-small.png" border="0" alt="Nagios XI" title="Nagios XI"></a>
</div>
<div id="pagetopalertcontainer">
</div>
<div id="authinfo">
</div>


<div id="topmenucontainer">
	<ul class="menu">
	<li><a href="http://10.1.1.111/nagiosxi/login.php">Login</a></li>
	</ul>
</div>
	<div id="feedback_layer">
	<div id="feedback_content">
	
	<div id="feedback_close">
	<a id="close_feedback_link" href="#"><img src="http://10.1.1.111/nagiosxi/images/b_close.png" border="0" alt="Close" title="Close"> Close</a>
	</div>
	
	<div id="feedback_container">
	
	<div id="feedback_header">
	<b>Send Us Feedback</b>
	<p>We love input!  Tell us what you think about this product and you'll directly drive future innovation!</p>
	</div><!-- feedback_header -->
	
	<div id="feedback_data">

	<form id="feedback_form" method="get" action="http://10.1.1.111/nagiosxi/ajaxproxy.php">

	<input type="hidden" name="proxyurl" value="http://api.nagios.com/feedback/">
	<input type="hidden" name="proxymethod" value="post">

	<input type="hidden" name="product" value="nagiosxi">
	<input type="hidden" name="version" value="2012R2.9">
	<input type="hidden" name="build" value="20140211">

	<label for="feedbackCommentBox">Comments:</label><br class="nobr" />
	<textarea class="textarea" name="comment" cols="40" rows="3"></textarea><br class="nobr" />

	<label for="feedbackNameBox">Your Name (Optional):</label><br class="nobr" />
	<input type="text" size="30" name="name" id="feedbackNameBox" value="" class="textfield" /><br class="nobr" />

	<label for="feedbackEmailAddressBox">Your Email Address (Optional):</label><br class="nobr" />
	<input type="text" size="30" name="email" id="feedbackEmailAddressBox" value="" class="textfield" /><br class="nobr" />

	<div id="feedbackFormButtons">
	<input type="submit" class="submitbutton" name="submitButton" value="Submit" id="submitFeedbackButton">
	</div>
	
	<br clear="all">
	<p>
	<a href="http://www.nagios.com/legal/privacypolicy/" target="_blank">Privacy Policy</a>
	</p>

	</form>
	
	</div><!-- feedback_data -->
	
	</div><!-- feedback_container-->
	
	
	</div><!--feedback_content-->
	</div><!--feedback_layer-->
<div id="popup_layer">
	<div id="popup_content">
	<div id="popup_close">
	<a id="close_popup_link" href="#"><img src="http://10.1.1.111/nagiosxi/images/b_close.png" border="0" alt="Close" title="Close"> Close</a>
	</div>
	<div id="popup_container">
	</div>
	</div>
</div>

<!-- HEADER END -->

		<div id="throbber"></div>
		</div><!--header -->


	<div id="mainframe">
	<div id="parentcontentthrobber"><img src='http://10.1.1.111/nagiosxi/images/throbber1.gif' /></div>
	<h1>Login</h1>

	<div style="float: left; margin-right: 25px; width: 50%;">



	<form id="loginForm" method="post" action="/nagiosxi/login.php">
	<input type='hidden' name='nsp' value='fb3798156288cc818e648fbf2725464a'>	
	<fieldset>
	<legend>Login</legend>
	
	<input type="hidden" name="page" value="auth">
	<input type="hidden" name="debug" value="">
	<input type="hidden" name="pageopt" value="login">

	<label for="usernameBox">Username:</label><br class="nobr" />
	<input type="text" size="10" name="username" id="usernameBox" class="textfield" /><br class="nobr" />
	
	<label for="passwordBox">Password:</label><br class="nobr" />
	<input type="password" size="10" name="password" id="passwordBox" class="textfield" /><br class="nobr" />
	
	<div id="formButtons">
	<input type="submit" class="submitbutton" value="Login" id="loginButton" name="loginButton"><br class="nobr" />
	
	
	</div>
	
	</fieldset>
	</form>
	
	<script type="text/javascript" language="JavaScript">
	document.forms['loginForm'].elements['usernameBox'].focus();
	</script>
	
	<br class="nobr" />

	<p>
	<a href="?forgotpass">Forgot your password?</a>
	</p>
	
	<label>Select Language:</label>
	<br class="nobr" />
	<div id='languageopts'>
		<a href='?locale=en_US' class='locale-icon en_US' title='English'></a>
		<a href='?locale=de_DE' class='locale-icon de_DE' title='German'></a>
		<a href='?locale=es_ES' class='locale-icon es_ES' title='Spanish'></a>
		<a href='?locale=fr_FR' class='locale-icon fr_FR' title='French'></a>
		<a href='?locale=it_IT' class='locale-icon it_IT' title='Italian'></a>
		<a href='?locale=ko_KR' class='locale-icon ko_KR' title='Korean'></a>
		<a href='?locale=pt_PT' class='locale-icon pt_PT' title='Portuguese'></a>
		<a href='?locale=ru_RU' class='locale-icon ru_RU' title='Russian'></a>
		<a href='?locale=zh_CN' class='locale-icon zh_CN' title='Simplified Chinese'></a>
		<a href='?locale=zh_TW' class='locale-icon zh_TW' title='Traditional Chinese'></a>
        <a href='?locale=ja_JP' class='locale-icon ja_JP' title='Japanese'></a>
	</div>
	
	</div>

	<div style="float: left; width: 40%;">

	
	
	<img src="http://10.1.1.111/nagiosxi/images/loginsplash.png"><br clear="all">
	<h3>About Nagios XI</h3>
	<p>
	Nagios XI is an enterprise-class monitoring and alerting solution that provides organizations with extended insight of their IT infrastructure before problems affect critical business processes.  For more information on Nagios XI, visit 
	<a href="http://www.nagios.com/products/nagiosxi/" target="_blank">www.nagios.com/products/nagiosxi/</a>
	</p>
	<h3>Nagios Learning Opportunities</h3>
	<p>
	Learn about Nagios 
	<a href="http://www.nagios.com/services/training" target="_blank"><strong>training</strong></a>
	and <a href="http://www.nagios.com/services/certification" target="_blank">
	<strong>certification</strong></a>.
	</p>
	<p>
	Want to learn about how other experts are utilizing Nagios?  Don't miss your chance to attend the next	<a href="http://go.nagios.com/nwcna" target="_blank">&nbsp;<strong> Nagios World Conference</strong></a>.
	</p>
	<h3>Contact Us</h3>
	<p>
	Have a question or technical problem? Contact us today:	</p>
	<table border="0">
	<tr><td valign="top">Support:</td>
		<td><a href="http://support.nagios.com/forum/" target="_blank">Online Support Forum</a></td></tr>
	<tr><td valign="top">Sales:</td><td>Phone: (651) 204-9102
	<br />Fax: (651) 204-9103
	<br />Email: sales@nagios.com</td></tr>
	<tr><td valign="top">Web:</td>
	<td><a href="http://www.nagios.com/" target="_blank">www.nagios.com</a></td></tr>
	</table>
	

</div>

	
		</div><!--mainframe-->
	
	<!--	<div id="footer">  //there should only be one div with id of footer on any given page, moved to footer.inc.php  --> 
		<!-- FOOTER START -->
<div id="footer">


    <div id="footermenucontainer">
        <div id="footernotice">Nagios XI 2012R2.9  Copyright &copy; 2008-2021 <a href="http://www.nagios.com/" target="_blank">Nagios Enterprises, LLC</a>.</div>

        
        <ul class="footermenu">
            <li><a href="http://10.1.1.111/nagiosxi/about/">About</a></li>
            <li><a href="http://10.1.1.111/nagiosxi/about/?legal">Legal</a></li>
        </ul>
    </div>

    

<script type="text/javascript">
    
function get_tray_alert_content() {     
    var optsarr = {
        "func": "get_tray_alert_html",
        "args": ""
    }
    var opts = array2json(optsarr);
    get_ajax_data_with_callback("getxicoreajax", opts, "process_tray_alert_content");
}
        
function process_tray_alert_content(edata) {
    data = unescape(edata);
    $("#tray_alerter_popup_content").html(data);
    
    var status = $("#tray_alerter_status").html();
    $("#tray_alerter").html(status);
}

$(document).ready(function() {

    get_tray_alert_content();
        
    $("#tray_alerter").everyTime(30000, "timer-tray_alerter", function(i) {
        get_tray_alert_content();
    });

    $("#tray_alerter").click(function() {
        var vis = $("#tray_alerter_popup").css("visibility");
        if (vis == "hidden") {
            $("#tray_alerter_popup").css("visibility", "visible");
        } else {
            $("#tray_alerter_popup").css("visibility", "hidden");
        }
    });
        
});
</script>

    <div id="keepalive"></div>

</div> <!-- end footer div -->

<!-- FOOTER END -->	<!-- 	</div>  -->
	
	</div><!--page-->

<noframes>
<!-- This page requires a web browser which supports frames. --> 
<h2>Nagios XI</h2>
<p align="center">
<a href="http://www.nagios.com/">www.nagios.com</a><br>
Copyright (c) 2009-2012 Nagios Enterprises, LLC<br>
</p>
<p>
<i>Note: These pages require a browser which supports frames</i>
</p>
</noframes>

		
<script type='text/javascript'>

  var _gaq = _gaq || [];
  _gaq.push(['_setAccount', 'UA-2887186-1']);
  _gaq.push(['_setAllowLinker', true]);
  _gaq.push(['_trackPageview']);

  (function() {
    var ga = document.createElement('script'); ga.type = 'text/javascript'; ga.async = true;
    ga.src = ('https:' == document.location.protocol ? 'https://ssl' : 'http://www') + '.google-analytics.com/ga.js';
    var s = document.getElementsByTagName('script')[0]; s.parentNode.insertBefore(ga, s);
  })();	

</script>	

</body>

</html>

[-] Cannot reliably check exploitability. Unable to obtain the value of the `nsp_str` token required for authentication
[*] Scanned 1 of 1 hosts (100% complete)
[*] Auxiliary module execution completed
msf6 auxiliary(scanner/http/nagios_xi_scanner) > 
```
</p></details>